### PR TITLE
parser: Deprecate VECTOR<FLOAT>. Use VECTOR instead.

### PR DESCRIPTION
**Pull request by EricZequan**
### What problem does this PR solve?

Issue Number: ref #54245 

Problem Summary:

According to latest spec, we plan to use the following type names:

- `VECTOR (currently, we support VECTOR, VECTOR<FLOAT>)`
- `VECTOR64 (previously we plan to use VECTOR<DOUBLE>)`
- `VECTOR16`

This PR use `VECTOR` as the canonical type name, in places like `SHOW CREATE TABLE ...` in order to be compatible with the new spec. `VECTOR<FLOAT>` is still supported, silently. We will no longer officially expose this kind of usage.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/builtin_compare.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ var (
 	_ builtinFunc = &builtinLeastVectorFloat32Sig{}
 	_ builtinFunc = &builtinIntervalIntSig{}
 	_ builtinFunc = &builtinIntervalRealSig{}
+	_ builtinFunc = &builtinLeastVectorFloat32Sig{}
```
Comment (breezewish) [inline pkg/expression/builtin_compare.go:69]: Mistake?

**Comment by breezewish** [comment]
The rest LGTM

File: pkg/expression/builtin_compare.go (hunk 1 above)
Comment (EricZequan) [reply pkg/expression/builtin_compare.go:69]: This code should be included in pre PR. I don't know how it get here.
Anyway, I remove it.

**Review by breezewish — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**